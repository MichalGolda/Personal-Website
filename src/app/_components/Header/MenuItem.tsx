import { useState } from "react";

export type MenuItemProps = {
  href: string;
  name: string;
};

export default function MenuItem({ href, name }: MenuItemProps) {
  const [hover, setHover] = useState<boolean>(false);

  return (
    <li>
      <a
        className="font-medium hover:text-secondary duration-500 transition-all text-body decoration-none"
        href={href}
        onMouseEnter={() => setHover(true)}
        onMouseLeave={() => setHover(false)}
      >
        {name}
        <span
          className={`w-0 bg-transparent transition-all duration-500 h-0.5  block ${
            hover ? "!bg-primary !w-2/3" : ""
          }`}
        ></span>
      </a>
    </li>
  );
}
