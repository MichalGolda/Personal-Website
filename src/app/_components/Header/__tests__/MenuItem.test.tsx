import "@testing-library/jest-dom"
import { fireEvent, render, screen } from "@testing-library/react"
import MenuItem from "../MenuItem"

describe("MenuItem component", () => {
    it("renders without errors", () => {
        const mockedOnClick = jest.fn()

        const { container } = render(<MenuItem href="/test" current={false} name="test" onClick={mockedOnClick} />)

        const item = container.children[0]
        const link = screen.getByRole("link")
        const decorator = link.children[0]

        expect(item).toBeInTheDocument()
        expect(item.tagName).toBe("LI")

        fireEvent.click(item)

        expect(mockedOnClick).toHaveBeenCalled()

        expect(link).toHaveTextContent("test")
        expect(link).toHaveAttribute("href", "/test")

        expect(decorator).not.toHaveClass("!bg-primary")

        fireEvent.mouseEnter(link)

        expect(decorator).toHaveClass("!bg-primary")

        fireEvent.mouseLeave(link)

        expect(decorator).not.toHaveClass("!bg-primary")
    })

    it("current highlight", () => {
        render(<MenuItem href="/test" current={true} name="test" onClick={() => {}} />)
        
        const link = screen.getByRole("link")
        const decorator = link.children[0]

        expect(decorator).toHaveClass("!bg-primary")
    })

    it("renders unchanged", () => {
        const { container } = render(<MenuItem href="/test" current={false} name="test" onClick={() => {}} />)
        
        expect(container).toMatchSnapshot()
    })
})