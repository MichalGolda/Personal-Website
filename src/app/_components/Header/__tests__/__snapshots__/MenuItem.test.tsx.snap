// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component renders unchanged 1`] = `
<div>
  <li>
    <a
      class="text-5xl font-medium hover:text-secondary duration-500 transition-[color] lg:text-base decoration-none"
      href="/test"
    >
      test
      <span
        class="w-0 bg-transparent transition-all duration-500 h-0.5  block "
      />
    </a>
  </li>
</div>
`;
